tests/trsapssam

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash -e
+# Copyright (C) 2022 Simo Sorce <simo@redhat.com>
+# SPDX-License-Identifier: Apache-2.0
+
+source "${TESTSSRCDIR}/helpers.sh"
+
+#Need to enable this or genpkey fails because it can't output the private key
+sed -e "s/#pkcs11-module-encode-provider-uri-to-pem/pkcs11-module-encode-provider-uri-to-pem = true/" \
+    "${OPENSSL_CONF}" > "${OPENSSL_CONF}.rsapss_genpkey"
+OPENSSL_CONF=${OPENSSL_CONF}.rsapss_genpkey
+
+ossl '
+genpkey -propquery "?provider=pkcs11"
+        -algorithm "RSA-PSS" -pkeyopt "rsa_pss_keygen_md:SHA256"
+        -pkeyopt "pkcs11_uri:pkcs11:object=Test-RSA-PSS-Restrictions"'
```

Comment:
This tests the case where we have explicit hash. I would prefer to have also test generating a key without the explicit hash and make sure the key can not be used for RSA PKCS#1.5 signature.

Reply:
Yeah I will add that and respin